### PR TITLE
Add web_startup_analyzer to material_3_demo

### PR DESCRIPTION
**Pull request by johnpryan**
This was mistakenly added to `experimental/material_3_demo`, not `material_3_demo` in https://github.com/flutter/samples/pull/2144

### CODE REVIEW

File: material_3_demo/web/index.html [hunk 1]
```diff
@@ -38,22 +38,30 @@
...
-          serviceWorkerVersion: serviceWorkerVersion,
-        },
-        onEntrypointLoaded: function(engineInitializer) {
-          engineInitializer.initializeEngine().then(function(appRunner) {
-            appRunner.runApp();
-          });
-        }
-      });
+<script>
+  var flutterWebStartupAnalyzer = new FlutterWebStartupAnalyzer();
```
Comment (kevmoo) [inline material_3_demo/web/index.html:45]: This rename feels...weird. Could we just have one variable?

Comment (johnpryan) [reply material_3_demo/web/index.html:45]: Good catch, I don't know why I wrote it like this.

**Comment by johnpryan** [comment]
I'm not sure we need experimental/material_3_demo anymore, so I submitted a separate PR to remove it: https://github.com/flutter/samples/pull/2153

File: material_3_demo/web/index.html [hunk 2]
```diff
@@ -38,22 +38,29 @@
...
+    _flutter.loader.loadEntrypoint({
+      serviceWorker: {
+        serviceWorkerVersion: serviceWorkerVersion,
+      },
+      onEntrypointLoaded: function(engineInitializer) {
+        analyzer.markFinished("loadEntrypoint");
+        analyzer.markStart("initializeEngine");
+        engineInitializer.initializeEngine().then(function(appRunner) {
+          analyzer.markFinished("initializeEngine");
+          analyzer.markStart("appRunnerRunApp");
```
Comment (kevmoo) [inline material_3_demo/web/index.html:58]: Is `"appRunnerRunApp"` special? Is it wired into the framework?

**Comment by kevmoo** [comment]
> I'm not sure we need experimental/material_3_demo anymore, so I submitted a separate PR to remove it: #2153

I've been building on it...but I don't know the difference either!

**Comment by domesticmouse** [comment]
> I'm not sure we need experimental/material_3_demo anymore, so I submitted a separate PR to remove it: #2153

Closed that PR as the `experimental` version was explicitly championed for by @HansMuller to enable publishing Material 3 content for APIs that are only in `master`.

File: material_3_demo/web/index.html (hunk 1 above)
Comment (johnpryan) [reply material_3_demo/web/index.html]: Actually this is needed as a global variable so that the Dart code can talk to it.

**Comment by HansMuller** [comment]
FTR, I think the time has come to remove experimental/material_3_demo. I suppose we should have taken care of that promptly after the last stable release - after M3 became the default - but now is still good.

CC @QuncCccccc

**Comment by johnpryan** [comment]
@kevmoo CI is passing, this is ready for another look.

File: material_3_demo/web/index.html (hunk 2 above)
Comment (kevmoo) [reply material_3_demo/web/index.html:59]: @johnpryan – thoughts here?

Comment (johnpryan) [reply material_3_demo/web/index.html:59]: Sorry, I missed this comment. These labels are arbitrary - they're only used to call into the web's performance API to mark timing information into the Performance timeline.

Comment (kevmoo) [reply material_3_demo/web/index.html:59]: is `appRunnerRunApp` ever marked finished? I didn't see it

File: web/_packages/web_startup_analyzer/lib/web_startup_analyzer.dart [hunk 3]
```diff
@@ -2,96 +2,5 @@
...
-    return completer.future;
-  }
-
-  Future<List<int>> captureFlutterFrameData() async {
-    await _frameAnalyzer.captureAdditionalFrames();
-    return _frameAnalyzer.additionalFrameTimes;
-  }
-}
+export 'src/web_startup_analyzer_io.dart'
+    if (dart.library.html) 'src/web_startup_analyzer.dart';
```
[the remark this answers is not in the document]
Comment (johnpryan) [reply web/_packages/web_startup_analyzer/lib/web_startup_analyzer.dart]: 😱 good catch

File: material_3_demo/web/index.html (hunk 2 above)
Comment (johnpryan) [reply material_3_demo/web/index.html:59]: It's marked as finished on the Dart side, when the`WebStartupAnalyzer` class is constructed in `main()`.

Comment (johnpryan) [reply material_3_demo/web/index.html:59]: There could be a smarter way to do this, this is just what I hacked together

Comment (johnpryan) [reply material_3_demo/web/index.html:59]: cc: @ditman

**Review by kevmoo — APPROVED** [review]

**Pull request merged**